cantools/autosar/end_to_end.py

```diff
@@ -37,7 +37,7 @@ def compute_profile2_crc(payload : Union[bytes, bytearray],
         assert msg.autosar.e2e.data_ids is not None
 
         protected_len = msg.autosar.e2e.payload_length
-        seq_counter = (payload[1]&0xf0) >> 4
+        seq_counter = payload[1] & 0xf
```

Comment:
are you sure that the other nibble is used  for the sequence counter? (I thought I tested this, but maybe I made a stupid mistake...)

Reply:
Yep, look [here](https://www.autosar.org/fileadmin/user_upload/standards/foundation/21-11/AUTOSAR_PRS_E2EProtocol.pdf) under `6.4.1 Header Layout`

Reply:
The high nibble is data:
![image](https://user-images.githubusercontent.com/52598363/195722778-cc92ceab-e7e2-4cf5-bdf1-787beab7fdc6.png)

Reply:
right. I suppose I only tested the code with data for profile 5 (which uses 16 bit sequence counters). for profile 2 I guess I became confused by the fact that `cantools dump` prints the highest significant bit on the left...